tests/calico_cni_k8s_test.go

```diff
@@ -52,6 +54,25 @@ func ensureNamespace(clientset *kubernetes.Clientset, name string) {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+func ensurePodDeleted(clientset *kubernetes.Clientset, ns string, podName string) {
+	err := clientset.CoreV1().Pods(ns).Delete(podName, metav1.NewDeleteOptions(0))
+	if kerrors.IsNotFound(err) {
+		return
+	}
+	if err != nil {
```

Comment:
Probably better to `Expect(err).NotTo(BeNil())`

If we can't do that here, then we'll want to return the error and have the calling code handle it.

Reply:
Updated code to do a `Get` first. And expect we do not have errors after `Delete`.

Reply:
I think either way we don't want `panics()` in the test code if we can avoid them - they make it a lot harder to read / understand test output and end the test short rather than running all to completion.